pandas/core/groupby/groupby.py

```diff
@@ -5829,6 +5808,66 @@ def func(df):
                     stacklevel=find_stack_level(),
                 )
 
+        if axis == 1:
+            try:
+                if self.obj.ndim == 1:
+                    result = self._op_via_apply(how, skipna=skipna)
+                else:
+
+                    def func(df):
+                        method = getattr(df, how)
+                        return method(
+                            axis=axis, skipna=skipna, numeric_only=numeric_only
+                        )
+
+                    func.__name__ = how
+                    result = self._python_apply_general(
+                        func, self._obj_with_exclusions, not_indexed_same=True
+                    )
+            except ValueError as err:
+                name = "argmax" if how == "idxmax" else "argmin"
+                if f"attempt to get {name} of an empty sequence" in str(err):
+                    raise ValueError(
+                        f"Can't get {how} of an empty group due to unobserved "
+                        "categories. Specify observed=True in groupby instead."
+                    ) from None
+                raise
+            return result
+
+        result = self._agg_general(
+            numeric_only=numeric_only,
+            min_count=1,
+            alias=how,
+            skipna=skipna,
+        )
+        return result
+
+    def _wrap_idxmax_idxmin(self, res: NDFrameT) -> NDFrameT:
+        index = self.obj._get_axis(self.axis)
+        if res.size == 0:
+            result = res.astype(index.dtype)
+        else:
+            if isinstance(index, MultiIndex):
+                index = index.to_flat_index()
+            values = res._values
+            if isinstance(values, ExtensionArray):
+                values = values.to_numpy(copy=False)
```

Comment:
Yes - for SeriesGroupBy only though. It's because of this:

```
print(type(pd.DataFrame({'a': [1, 1, 2]}, dtype="Int64")._values))
# <class 'numpy.ndarray'>

print(type(pd.Series([1, 1, 2], dtype="Int64")._values))
# <class 'pandas.core.arrays.integer.IntegerArray'>
```

Reply:
Though it was hit, I believe it can be safely removed as long as we are okay passing the IntegerArray to `index.array.take` in the Series path below.

Reply:
Looks like .take immediately converts it to an ndarray in `ensure_platform_int` anyways. Prefer to have this removed?

Reply:
my intuition is that if we get here with an EA then something has gone wrong.

Reply:
(im OK with merging and looking into this more closely later, as you've already put a lot of time into this)

Reply:
id guess that the place where something has gone wrong is in the MaskedEA._groupby_op not short-circuiting _maybe_mask_result

Reply:
Ah, I see. I didn't think of avoiding _maybe_mask_result - this makes sense to do.